extensions/amp-accordion/0.1/test/integration/test-amp-accordion.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe('amp-accordion', function() {
+  this.timeout(10000);
+  const extensions = ['amp-accordion'];
+  const body = `
+  <amp-accordion media="(min-width: 500px)" id="media-accordion">
+    <section>
+      <h1>
+        Title
+      </h1>
+      <p>Bunch of awesome content</p>
+    </section>
+  </amp-accordion>
+  `;
+  describes.integration('amp-accordion', {
+    body,
+    extensions,
+  }, env => {
+
+    let win, iframe, doc;
+    beforeEach(() => {
+      win = env.win;
+      iframe = env.iframe;
+      doc = win.document;
+      iframe.width = 300;
+    });
+
+    it('should respect the media attribute', () => {
+      const accordion = doc.getElementById('media-accordion');
+      expect(iframe.clientWidth).to.equal(300);
+      expect(accordion.className).to.match(/i-amphtml-hidden-by-media-query/);
+      iframe.width = 600;
+      expect(iframe.clientWidth).to.equal(600);
+      return timeout(200).then(() => {
```

Comment:
why not `poll`?

Reply:
Because there was just one expectation here, so if this is used as the polling condition, there's nothing more to expect after the poll? 

`expect(accordion.className).to.not.match(/i-amphtml-hidden-by-media-query/);`

Reply:
@rsimha would you by chance have any better ideas? Basically I'm trying to test a media query after a resize. So after the resize, we need to wait for the runtime to eventually go through the applySizesAndMedia logic and then toggle the hidden-by-media-query class before verifying it. Is there a good way within the test framework to wait for that and then verify?

Reply:
`poll` would have an implicit expect, there isn't really a need for explicit one. `return poll('hidden by media query', () => {...}, `Was not hidden by media query but should have`);`

would fail the test with `Was not hidden by media query but should have` if condition is not satisfied.

Reply:
Another way to approach this is to stub / mock the function you expect to get called, and verify an expectation that it did get called with some expected value.

Reply:
> Is the root of the problem the fact that the runtime is inherently asynchronous and doesn't signal event completion?
Precisely. 
> Another way to approach this is to stub / mock the function you expect to get called, and verify an expectation that it did get called with some expected value.
So like the waitForAttribute function that @cvializ introduces in 
https://github.com/ampproject/amphtml/pull/16194?

Reply:
these are integration tests, we should stay away from mocking or calling any JS methods or expecting any internal events.

Reply:
Fair point re: integration tests. If these are DOM elements you're waiting to be modified, would it help to use something like http://www.chaijs.com/plugins/chai-dom?

Reply:
There's also this: http://www.chaijs.com/plugins/chai-change

Reply:
Since the values are set asynchronously, I don't think that `chai-dom` will help either. Unfortunately `poll` seems to cause flakes in many cases.

Reply:
`chai-dom` doesn't wait and is sync. There are three options:

1- poll
2- mutation observer
3- mock

poll is the most realistic one, same way user has to wait to see something, we also wait to see it reflected in the dom. we need to figure out why `poll` causes flakes in cases like #16194, it should not if the `condition` is permanent. Only case `poll` should cause flakes is if the condition we are waiting for is temporarily and can change within the `8ms` gaps. That should be rare and assertion of temporary state probably should not be in integration tests anyway.

mutation observer can replace poll in certain cases only and is not supported in all browsers. In theory switching from poll to mutation observer should not save more than 10ms per call.

mock: we should not do it.

Reply:
(or better 4- we use proper E2E testing tools instead of our own integration testing framework: https://github.com/ampproject/amphtml/pull/16194 )

Reply:
we gotta figure out why `poll` causes flakes. Is it just certain platforms? position-observer integration tests use `poll` extensively and have never flaked (but they only run in `newChrome`)

Reply:
Checking now on windows platform. I was able to repro locally in windows chrome.